File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int>ans(temperatures.size(),0);
        stack<int>stack;
        for(int i=0;i<temperatures.size();i++){
            while(!stack.empty() && temperatures[stack.top()] < temperatures[i]){
                int index = stack.top();
                stack.pop();
                ans[index]=i-index;
            }
            stack.push(i);
        }
        return ans;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer>stack = new Stack<>();
        int[] arr = new int[temperatures.length];
        Arrays.fill(arr,0);
        for(int i=0;i<temperatures.length;i++){
            while(!stack.isEmpty() && temperatures[i]>temperatures[stack.peek()]){
                arr[stack.peek()] = i - stack.peek();
                stack.pop();
            }
            stack.push(i);
        }
        return arr;
    }
}